File: 762/prime-number-of-set-bits-in-binary-representation.py
class Solution:
    def countPrimeSetBits(self, L, R):
        """
        :type L: int
        :type R: int
        :rtype: int
        """
        count = 0
        for i in range(L, R+1):
            weight = self.hamming(i)
            if self.is_prime(weight):
                count += 1
        return count

    def hamming(self, n):
        res = 0
        while n > 0:
            tmp = n % 2
            if tmp == 1:
                res += 1
            n = (n - tmp) >> 1
        return res
    
    # speed up
    # def hamming(self, n):
    #     res = 0
    #     while n != 0:
    #         res += 1
    #         n = n & (n - 1)
    #     return res

    def is_prime(self, n):
        if n < 2:
            return False
        if n == 2 or n == 3:
            return True
        for i in range(2, n):
            if n % i == 0:
                return False
        return True

    # speed up
    # def is_prime(self, n):
    #     prime_set = set([2, 3, 5, 7, 11, 13, 17, 19])
    #     if n in prime_set:
    #         return True
    #     return False

File: 762/prime-number-of-set-bits-in-binary-representation.js
/**
 * @param {number} L
 * @param {number} R
 * @return {number}
 */
var countPrimeSetBits = function (L, R) {
    let count = 0;
    for (let i = L; i <= R; i++) {
        let weight = hammingWeight(i);
        if (isPrime(weight))
            count++;
    }
    return count;
};

var hammingWeight = function (n) {
    let count = 0;
    while (n > 0) {
        let tmp = n % 2;
        if (1 == tmp) {
            count++;
        }
        n = (n - tmp) / 2;
    }
    return count;
};

var hammingWeight = function (n) {
    let count = 0;
    while (n != 0) {
        count++;
        n = n & (n - 1);
    }
    return count;
};

var isPrime = function (n) {
    if (n < 2) return false;
    if (n == 2 || n == 3) return true;
    for (let i = 2; i < Math.sqrt(n) + 1; i++) {
        if (n % i == 0)
            return false;
    }
    return true;
};

var isPrime = function (n) {
    const primeSet = new Set([2, 3, 5, 7, 11, 13, 17, 19]);
    if (primeSet.has(n)) {
        return true;
    }
    return false;
};